src/github.ts

```diff
@@ -0,0 +1,140 @@
+ * @param assignees The list of assignees to assign the PR.
+ */
+export async function setAssignees(assignees: string[]): Promise<void> {
+  const pr = getPr().number;
+  core.info(`Setting Assignees for PR #${pr} to: ${assignees.join(',')}`);
+
+  const response = await client.rest.issues.addAssignees({
+    owner: github.context.repo.owner,
+    repo: github.context.repo.repo,
+    issue_number: pr,
+    assignees: assignees,
+  });
+  coreDebugJson(response, `setAssignees(${pr}, [${assignees.join(',')}]) > response`);
+}
+
+export type CheckRun = components['schemas']['check-run'];
+
+/**
+ * Get the checkruns for the current context's workflow_run for the head SHA.
+ */
+export async function getCheckRuns(): Promise<CheckRun[]> {
+  const ref = github.context.payload.workflow_run.head_sha;
+  const { owner, repo } = github.context.repo;
+
+  core.info(`Retrieving check runs for ${owner}/${repo}@${ref}...`);
+  const checkRuns: CheckRun[] = (
+    await client.rest.checks.listForRef({
+      owner,
+      repo,
+      ref,
+    })
+  ).data.check_runs;
+
+  core.info(`Retrieved ${checkRuns.length} check runs.`);
+  coreDebugJson(checkRuns, 'getCheckRuns() > checkRuns');
+
+  return checkRuns;
+}
+
+/**
+ * Get the required checks for the base ref.
+ */
+export async function getRequiredCheckNames(): Promise<string[] | undefined> {
+  const baseRef = getPr()?.base.ref;
+  core.info(`Base Ref: ${baseRef}`);
+  if (baseRef === undefined) {
+    core.error(`Error getting base ref for PR #${getPr()?.number}.`);
+    return undefined;
+  }
+  const { owner, repo } = github.context.repo;
+
+  core.info(`Retrieving branch protection information for ${owner}/${repo}@${baseRef}...`);
+  const result: components['schemas']['status-check-policy'] = (
+    await client.rest.repos.getStatusChecksProtection({
+      owner,
+      repo,
+      branch: baseRef,
+    })
+  ).data;
+
+  coreDebugJson(result, 'getRequiredChecks() > result');
+
+  return result?.contexts;
+}
+
+/**
+ * Get user inputs as an array (expects the user input to be CSV)
+ * @param name The name of the user input
+ * @param defaultVal The default value
+ */
+export function getInputArray(name: string, defaultVal: string[]): string[] {
+  return (
+    core
+      .getInput(name)
+      .split(',')
+      .map((i) => i.trim()) || defaultVal
+  );
+}
+
+/**
+ * Get user inputs
+ * @param name The name of the user input
+ * @param defaultVal The default value
+ */
+export function getInput(name: string, defaultVal: string | null): string | null {
+  return core.getInput(name) || defaultVal;
+}
+
+/**
+ * Helper to pretty-print JSON as log debug.
+ * @param value The object to pretty-print
+ * @param name The name (used for header/footer)
+ */
+export function coreDebugJson(value: object, name: string) {
+  core.debug(`====== BEGIN ${name} ======`);
+  core.debug(JSON.stringify(value, null, 4));
+  core.debug(`======= END ${name} =======`);
+}
```

Comment:
Not really understanding why you need this.

Reply:
Simply for debugging - I was making prolific use of this when testing the action against PRs I was opening. I just didn't want to keep repeating the same 3 lines over and over. Happy to remove if you feel it's not useful.